src/ompl/geometric/planners/rrt/src/STRRTstar.cpp

```diff
@@ -518,7 +518,14 @@ ompl::geometric::STRRTstar::GrowState ompl::geometric::STRRTstar::growTreeSingle
 
     /* find state to add */
     ompl::base::State *dstate = rmotion->state;
-    double d = si_->distance(nmotion->state, rmotion->state);
+
+    double d = 0;
+    if(tgi.start){
+        d = si_->distance(nmotion->state, rmotion->state);
+    }
+    else{
+        d = si_->distance( rmotion->state,nmotion->state);
+    }
```

Comment:
3) in function ompl::geometric::STRRTstar::growTreeSingle, when we are growing from start tree, nmotion->state is "past", and rmotion->state is "future", so to satisfy t1<t2, we must pass this arguments as si_->distance(nmotion->state, rmotion->state);

When we are growing from goal tree, nmotion->state is "future", and rmotion->state is "past", so to satisfy t1<t2, we must pass this arguments as si_->distance( rmotion->state,nmotion->state);